### DISCUSSION TITLE
Crashes from Crashlytica

### DISCUSSION THREAD

**Discussion by fury21** [LeoNatan/LNPopupController#558] [General] [answered]
Hi.
I started receiving multiple crash reports in Firebase Crashlytics

Fatal Exception: NSInternalInconsistencyException
Modifications to the layout engine must not be performed from a background thread after it has been accessed from the main thread.

Fatal Exception: NSInternalInconsistencyException
0  CoreFoundation                 0xec69c __exceptionPreprocess
1  libobjc.A.dylib                0x2bc80 objc_exception_throw
2  CoreAutoLayout                 0x56c8 NSISSparseVectorAddTermWithPlaceValueCoefficientStartingIndex
3  CoreAutoLayout                 0xa4e4 -[NSISEngine withBehaviors:performModifications:]
4  UIKitCore                      0x50ac8 __57-[UIView(AdditionalLayoutSupport) _switchToLayoutEngine:]_block_invoke
5  UIKitCore                      0x508a4 -[UIView(AdditionalLayoutSupport) _switchToLayoutEngine:]
6  UIKitCore                      0x4e944 -[UIView(AdditionalLayoutSupport) _initializeHostedLayoutEngine]
7  UIKitCore                      0x4dae8 -[UIView(AdditionalLayoutSupport) _layoutEngineCreateIfNecessary]
8  UIKitCore                      0x4d994 __120-[UIView(UIConstraintBasedLayout) _layoutEngine_didAddLayoutConstraint:roundingAdjustment:mutuallyExclusiveConstraints:]_block_invoke_2
9  UIKitCore                      0x4d584 -[UIView _layoutEngine_didAddLayoutConstraint:roundingAdjustment:mutuallyExclusiveConstraints:]
10 UIKitCore                      0x4d378 -[UIView _tryToAddConstraint:roundingAdjustment:mutuallyExclusiveConstraints:]
11 UIKitCore                      0x4d258 -[UIView(UIConstraintBasedLayout) nsli_addConstraint:]
12 CoreAutoLayout                 0xa284 +[NSLayoutConstraint _addOrRemoveConstraints:activate:]
13 UIKitCore                      0x12b2f0 -[UINavigationController _setUpContentFocusContainerGuide]
14 UIKitCore                      0x12b074 -[UINavigationController loadView]
15 UIKitCore                      0x15548 -[UIViewController loadViewIfRequired]
16 UIKitCore                      0x14048 -[UIViewController view]
17 UIKitCore                      0xcaaa4 -[UINavigationController pushViewController:transition:forceImmediate:]
18 UIKitCore                      0xca6f4 -[UINavigationController pushViewController:animated:]
19 Sofy 2.0                       0x536418 -[UINavigationController(LNPopupSupportPrivate) _ln_pushViewController:animated:] + 1150 (UIViewController+LNPopupSupportPrivate.m:1150)
20 UIKitCore                      0xc9634 -[UINavigationController initWithRootViewController:]
21 Sofy 2.0                       0xa1034 MainCoordinator.mainTabBar(showFullAccessVC:) + 420 (MainCoordinator.swift:420)
22 Sofy 2.0                       0x4f4e4 closure #1 in ReminderViewController.remindButtonAction() + 235 (ReminderViewController.swift:235)
23 Sofy 2.0                       0xcee0c closure #1 in ScheduleManager.registerForNotifications(completion:) + 184 (ScheduleManager.swift:184)
24 Sofy 2.0                       0x53b2c thunk for @escaping @callee_guaranteed (@unowned Bool, @guaranteed Error?) -> () (<compiler-generated>)
25 libdispatch.dylib              0x26a8 _dispatch_call_block_and_release
26 libdispatch.dylib              0x4300 _dispatch_client_callout
27 libdispatch.dylib              0xb894 _dispatch_lane_serial_drain
28 libdispatch.dylib              0xc3f8 _dispatch_lane_invoke
29 libdispatch.dylib              0x17004 _dispatch_root_queue_drain_deferred_wlh
30 libdispatch.dylib              0x16878 _dispatch_workloop_worker_thread
31 libsystem_pthread.dylib        0x1964 _pthread_wqthread
32 libsystem_pthread.dylib        0x1a04 start_wqthread

97% of devices received this error on iOS 17
LNPopupController version 2.14

What does this error mean? How can it be fixed? If possible, without updating to the latest version of the framework

**Comment by iDevelopper** [comment, marked as answer]
It looks like you need to update your app's UI on the main thread!

**Reply by fury21** [reply]
But in which place?
